src/time/Storage.cpp

```diff
@@ -148,7 +163,19 @@ Eigen::VectorXd Storage::sample(double normalizedDt) const
 
   auto data = getTimesAndValues();
 
-  return math::bspline::interpolateAt(data.first, data.second, usedDegree, normalizedDt);
+  //If t equals to one timestep in data.first then we want to return the corresponding sample
+  for (int i = 0; i < data.first.size(); i++) {
+    if (math::equals(data.first[i], normalizedDt)) {
+      return data.second.col(i);
+    }
+  }
```

Comment:
@NiklasKotarsky you are referring to this part resolving #1752, correct? @fsimonis and I have some concerns w.r.t possible jumps in the interpolant. Is this part here really necessary to solve your performance problem or is it not really relevant? If there is no clear performance gain, I would leave it out in favor of a simpler and more consistent implementation. But you probably know the pros and cons best.

Reply:
Yes, after some thought I would also say that this is not going to matter from a performance perspective, since one could always use 0:th order interpolation.